modules/libcom/src/iocsh/iocsh.cpp

```diff
@@ -571,8 +576,25 @@ char** iocsh_attempt_completion(const char* word, int start, int end)
             // known argument type
             rl_attempted_completion_over = 1;
 
+            //if input string has a ., we know what record it is.
+            std::string str = word;
+            std::string init_field = "";
+            std::string pv = "";
+
+            size_t dot = str.find(".");
+
+            if (dot != std::string::npos) {
+                pv = str.substr(0, dot);
+                init_field = str.substr(dot+1u, str.length());
+            }
+
+            if(!init_field.empty()) {
+                return iocshCompleteField(pv.c_str(), init_field.c_str());
+            }
+
+            //Q: where is iocshCompleteRecord instantiated?
```

Comment:
Apparently it comes from here: https://github.com/epics-base/epics-base/blob/5eff3803a83fbc3ea855b63841f5359c09830ade/modules/database/src/ioc/db/dbIocRegister.c#L506

https://github.com/epics-base/epics-base/blob/5eff3803a83fbc3ea855b63841f5359c09830ade/modules/database/src/ioc/dbStatic/dbCompleteRecord.cpp#L90